sites/partners/src/listings/PaperListingForm/UnitForm.tsx

```diff
@@ -78,6 +80,17 @@ const UnitForm = ({ onSubmit, onClose, defaultUnit, nextId, draft }: UnitFormPro
     name: "amiChart.id",
   })
 
+  const minOccupancy: number = useWatch({
+    control,
+    name: "minOccupancy",
+  })
+
+  const maxOccupancy: number =
+    useWatch({
+      control,
+      name: "maxOccupancy",
+    }) || numberOccupancyOptions
```

Comment:
I'm not sure this or operation here is the best way to do this, useWatch acts like other hooks, so you're setting maxOccupancy to be the result of a hook or 10.

I think instead the or should happen where the where the comparison is needed, so in this case lines ~459

Reply:
@YazeedLoonat this useWatch is only being used as validation for the minOccupancy field. The number of options in the Select remains constant regardless. So if the user has selected a minOccupancy but not a maxOccupancy, without this, the form throws an error, because minOccupancy > undefined. This 'or' is only to say if nothing has been selected in MaxOccupancy, don't throw an error unless minOccupancy is greater than 10 (which isn't possible in the dropdown).

Reply:
@YazeedLoonat the first thing I tried was `validation={maxOccupancy && { max: maxOccupancy }}`, which threw console warnings. I then tried `validation={{ max: maxOccupancy || numberOccupancyOptions}}` which also threw console warnings. When I moved it here, they went away. I think because `register(validation)` doesn't want conditional statements passed into it. I figured it would be better to handle the logic before it hit the register function.